include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -7511,14 +7511,12 @@ FEFaceEvaluation<dim,
               this->face_numbers[i] = faces.exterior_face_no;
             }
 
-          const bool   orientation_interior_face = faces.face_orientation >= 8;
-          unsigned int face_orientation          = faces.face_orientation % 8;
+          const bool orientation_interior_face = faces.face_orientation >= 8;
+          auto       face_orientation          = faces.face_orientation % 8;
           if (face_identifies_as_interior != orientation_interior_face)
-            {
-              constexpr std::array<std::uint8_t, 8> table{
-                {0, 1, 6, 3, 4, 5, 2, 7}};
-              face_orientation = table[face_orientation];
-            }
+            face_orientation =
+              ReferenceCells::get_hypercube<dim - 1>()
+                .get_inverse_combined_orientation(face_orientation);
```

Comment:
Is there an easy way to get the reference cell in this context? AFAICT that requires getting a quadrature index from MappingInfo, but we don't have that index here.

Reply:
That's a good question. I guess it depends on whether we'd use it for an assertion (where we could ask `this->matrix_free->get_cell_iterator(cell, i)->face(face_number)->reference_cell())`), or if it runs unconditionally also in release mode. I have no good feeling how expensive it would be. But we could just do what you think is right for now, and in case it does hurt us, we can come back to it?